### PR TITLE
Move ConfigMap to main API

### PR DESCRIPTION
**Pull request by pmorie**
Moves the ConfigMap resource to the main API from extensions.

@bgrant0607 @thockin @erictune @smarterclayton 
cc @tmrts

### CODE REVIEW

**Comment by bgrant0607** [comment]
I agree with the move, if someone else can look at the details.

**Comment by smarterclayton** [comment]
LGTM

**Comment by thockin** [comment]
LGTM

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -1972,6 +1972,36 @@ func ValidateSecretUpdate(newSecret, oldSecret *api.Secret) field.ErrorList {
...
+// ValidateConfigMap tests whether required fields in the ConfigMap are set.
+func ValidateConfigMap(cfg *api.ConfigMap) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateObjectMeta(&cfg.ObjectMeta, true, ValidateConfigMapName, field.NewPath("metadata"))...)
+
+	for key := range cfg.Data {
+		if !IsSecretKey(key) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, fmt.Sprintf("must have at most %d characters and match regex %s", validation.DNS1123SubdomainMaxLength, SecretKeyFmt)))
+		}
+	}
```
Comment (thockin) [inline pkg/api/validation/validation.go]: Should we fold in the size-checking into this PR or followup?

**Comment by thockin** [comment]
What is the shortcut name for configmap?  "conf" or "cm" or "cfg" or "cfm" or ... ?

**Comment by thockin** [comment]
Assuming that the size-checking will be a followup, I am LGTMing this to maybe keep the flow moving.

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by thockin** [comment]
@k8s-bot unit test this please

**Comment by pmorie** [comment]
@thockin I wanted to keep the follow-ups as separate PRs because this is a chunk to review on its own.

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (pmorie) [inline pkg/api/validation/validation.go]: Follow-up, this is enough on its own.

**Comment by pmorie** [comment]
@thockin 

> What is the shortcut name for configmap? "conf" or "cm" or "cfg" or "cfm" or ... ?

I like 'cm', will add in a follow-up

**Comment by pmorie** [comment]
Re-adding LGTM after rebase.

**Comment by pmorie** [comment]
@k8s-bot test this

**Comment by pmorie** [comment]
@k8s-bot test this

**Comment by pmorie** [comment]
@k8s-bot test this

**Pull request merged**